bin/check_go_path

```diff
@@ -7,11 +7,6 @@ if [ -z "$PWD" ]; then
 	exit 1
 fi
 
-if [ -z "$GOPATH" ]; then
-	echo "GOPATH not set, you must have go configured properly to install ipfs"
-	exit 1
-fi
-
```

Comment:
I'd still keep this check (assuming it didn't break, which it shouldn't) so if something goes wrong it's caught.

Reply:
I disagree, we are already passing `GOPATHs` as arguments to this function (as `$2`, `$3` and so on). So this check is IMO not needed.

Reply:
GOPATH doesn't have to be set anymore. I'm actually not sure why the `export` in golang.mk doesn't carry over.

Reply:
It does, I just wanted to make this script be runnable semi stand alone.